compiler/rustc_trait_selection/src/traits/normalize.rs

```diff
@@ -66,7 +66,7 @@ impl<'tcx> At<'_, 'tcx> {
             let value = self
                 .normalize(value)
                 .into_value_registering_obligations(self.infcx, &mut *fulfill_cx);
-            let errors = fulfill_cx.select_where_possible(self.infcx);
+            let errors = fulfill_cx.select_all_or_error(self.infcx);
```

Comment:
why this old solver change? It doesn't seem to break anything, I am a bit worried that it may impact something however as we reuse the `fulfill_cx` of the caller

Reply:
I will add an assertion here that the fulfill_cx is empty

Reply:
assert that fulfill_cx is empty :3